### PR TITLE
Feature/edo 250 merge latest changes in otto repo

### PR DESCRIPTION
**Pull request by fbelter-iteratec**

### CODE REVIEW

File: src/main/webapp/app/teams/teams-skills.component.html [hunk 1]
```diff
@@ -53,73 +55,73 @@ <h4>{{'teamdojoApp.teams.skills.title' | translate}}</h4>
...
+                               (click)="setIncomplete(skill)">
+                                <fa-icon icon="check"></fa-icon>
+                                <small dojoTranslate="teamdojoApp.teams.skills.action.admin.incomplete"></small>
                             </a>
                         </div>
                         <div *ngIf="isVoteAble(skill)"
                              class="skill-action-bar d-flex flex-wrap justify-content-between">
                             <p class="skill-action-bar-item">
-                                <small dojoTranslate="teamdojoApp.teams.skills.action.vote" [translateValues]="{'team': team.name }"></small>
+                                <small dojoTranslate="teamdojoApp.teams.skills.action.vote"
```
Comment (sberlin) [inline src/main/webapp/app/teams/teams-skills.component.html]: The font is too big so the voting icons break onto a new line.

File: src/main/webapp/app/teams/teams-skills.component.html [hunk 2]
```diff
@@ -53,73 +55,73 @@ <h4>{{'teamdojoApp.teams.skills.title' | translate}}</h4>
...
-                            <a (click)="upVote(skill)">
-                                <fa-icon icon="angle-up"></fa-icon>
-                            </a></p>
                         </div>
                         <div *ngIf="isSuggestAble(skill)"
                              class="skill-action-bar d-flex flex-wrap justify-content-between">
                             <p class="skill-action-bar-item">
-                                <a (click)="suggest(skill)"><small dojoTranslate="teamdojoApp.teams.skills.action.suggest" [translateValues]="{'team': team.name }"></small></a>
+                                <a (click)="suggest(skill)">
+                                    <small dojoTranslate="teamdojoApp.teams.skills.action.suggest"
```
Comment (sberlin) [inline src/main/webapp/app/teams/teams-skills.component.html]: The font is too big.

File: src/main/webapp/app/teams/skill-details/skill-details-status/skill-details-status.scss [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        a {
+            small {
+                color: $dojo-text-secondary;
+            }
+        }
+        i.skill-relevance-action-item {
+            &.skill-status-irrelevant {
+                &:before {
+                    content: '\f164'; // thumbs-up
```
Comment (sberlin) [inline src/main/webapp/app/teams/skill-details/skill-details-status/skill-details-status.scss]: This approach tries to dynamically style action icons which reduces condition paths inside the html. Duplicating the style from font awesome is not optimal but I can accept this for better readability of the template.

File: src/main/webapp/app/shared/skill-details/skill-details-info.scss [hunk 4]
```diff
@@ -2,13 +2,15 @@
 @import '../../../content/scss/bootstrap-variables';
 @import '../../../content/scss/colors';
 @import '../../../content/scss/mixins';
+@import '../../teams/skill-details/skill-details-status/skill-details-status';
```
Comment (sberlin) [inline src/main/webapp/app/shared/skill-details/skill-details-info.scss]: This leads to tight coupling with the component within the teams module. When sharing code this should be extracted to a shared dependency.

File: src/main/java/de/otto/teamdojo/domain/Skill.java [hunk 5]
```diff
@@ -45,9 +45,8 @@
     @Column(name = "jhi_validation", length = 2048)
     private String validation;
 
-    @Pattern(regexp = "^P(?:([-+]?[0-9]+)Y)?(?:([-+]?[0-9]+)M)?(?:([-+]?[0-9]+)W)?(?:([-+]?[0-9]+)D)?$")
```
[the remark this answers is not in the document]
Comment (fbelter-iteratec) [reply src/main/java/de/otto/teamdojo/domain/Skill.java]: Regenerated Skill entity with generator.

File: src/main/java/de/otto/teamdojo/service/SkillQueryService.java [hunk 6]
```diff
@@ -103,7 +103,10 @@ public long countByCriteria(SkillCriteria criteria) {
                 specification = specification.and(buildStringSpecification(criteria.getValidation(), Skill_.validation));
             }
             if (criteria.getExpiryPeriod() != null) {
-                specification = specification.and(buildStringSpecification(criteria.getExpiryPeriod(), Skill_.expiryPeriod));
+                specification = specification.and(buildRangeSpecification(criteria.getExpiryPeriod(), Skill_.expiryPeriod));
+            }
+            if (criteria.getScore() != null) {
+                specification = specification.and(buildRangeSpecification(criteria.getScore(), Skill_.score));
```
[the remark this answers is not in the document]
Comment (fbelter-iteratec) [reply src/main/java/de/otto/teamdojo/service/SkillQueryService.java]: Removed duplicate

File: README.md [hunk 7]
```diff
@@ -199,3 +199,38 @@ Please change the password and secret in your production environment.
...
+
+Determining the status of a skill acquired by a Team was reworked.
+
+A combination of data from the Skill and Teamskill entities are used in the determination of the status.
+
+The fields used in calculating the status are "irrelevant" and "completedAt" (From the TeamSkill Entity) and "expiryPeriod" (from the Skill Entity).
+
+In the context of a given Team, a skill is in the 
+
+  OPEN Status - If the TeamSkill.completedAt field is not set (that is null) and TeamSkill.irrelevant is not true
```
[the remark this answers is not in the document]
Comment (fbelter-iteratec) [reply README.md]: Changed formatting

File: src/main/webapp/app/teams/teams-skills.component.html (hunk 1 above)
Comment (fbelter-iteratec) [reply src/main/webapp/app/teams/teams-skills.component.html]: Changed styling

File: src/main/webapp/app/teams/teams-skills.component.html (hunk 2 above)
Comment (fbelter-iteratec) [reply src/main/webapp/app/teams/teams-skills.component.html]: Changed styling

File: src/main/webapp/app/teams/skill-details/skill-details-status/skill-details-status.scss (hunk 3 above)
Comment (fbelter-iteratec) [reply src/main/webapp/app/teams/skill-details/skill-details-status/skill-details-status.scss]: Ok.

File: src/main/webapp/app/shared/skill-details/skill-details-info.scss (hunk 4 above)
Comment (fbelter-iteratec) [reply src/main/webapp/app/shared/skill-details/skill-details-info.scss]: Well seen! Moved styles from skill-details-status to skill-details-info. Styles were only used here.

File: src/test/java/de/otto/teamdojo/web/rest/SkillResourceIntTest.java [hunk 8]
```diff
@@ -62,8 +61,9 @@
     private static final String DEFAULT_VALIDATION = "AAAAAAAAAA";
     private static final String UPDATED_VALIDATION = "BBBBBBBBBB";
 
-    private static final String DEFAULT_EXPIRY_PERIOD = "P9M+71W-05D";
-    private static final String UPDATED_EXPIRY_PERIOD = "P25D";
+    private static final Integer DEFAULT_EXPIRY_PERIOD = 24;
+
+    private static final Integer UPDATED_EXPIRY_PERIOD = 21;
```
Comment (sberlin) [inline src/test/java/de/otto/teamdojo/web/rest/SkillResourceIntTest.java]: Please switch these two values to conform to the assertions regarding the updated value being greater than the default in `getAllSkillsByExpiryPeriodIsGreaterThanOrEqualToSomething` and `getAllSkillsByExpiryPeriodIsLessThanSomething`.

Comment (fbelter-iteratec) [reply src/test/java/de/otto/teamdojo/web/rest/SkillResourceIntTest.java]: Test fixed.

**Comment by fbelter-iteratec** [comment]
The provided links above ("deprecated toString()" & "duplicated setter") are taking me to the same as https://github.com/iteratec/iteratec-teamdojo/pull/75/files
Are there issues left, which i don't see?

**Comment by sberlin** [comment]
Try these links:
https://github.com/iteratec/iteratec-teamdojo/blob/a62cef24586c5c1cadcec8077cf4b123982c9531/src/test/java/de/otto/teamdojo/web/rest/SkillResourceIntTest.java#L271
https://github.com/iteratec/iteratec-teamdojo/blob/a62cef24586c5c1cadcec8077cf4b123982c9531/src/test/java/de/otto/teamdojo/web/rest/SkillResourceIntTest.java#L855

**Pull request merged**